From b2972c11495eafdb5ee16e08536a3d1a25b1b171 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:11:26 -0400
Subject: [ethernet] i40evf: remove open-coded skb_cow_head

Message-id: <1409663591-32531-44-git-send-email-sassmann@redhat.com>
Patchwork-id: 90518
O-Subject: [RHEL7.1 PATCH 043/148] i40evf: remove open-coded skb_cow_head
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Francois Romieu <romieu@fr.zoreil.com>

Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
Cc: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit fe6d4aa437baf185bc98b83eaff229a3ac5d738c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40e_txrx.c b/drivers/net/ethernet/intel/i40evf/i40e_txrx.c
index 53be5f4..b9f50f4 100644
--- a/drivers/net/ethernet/intel/i40evf/i40e_txrx.c
+++ b/drivers/net/ethernet/intel/i40evf/i40e_txrx.c
@@ -1114,20 +1114,18 @@ static int i40e_tso(struct i40e_ring *tx_ring, struct sk_buff *skb,
       u64 *cd_type_cmd_tso_mss, u32 *cd_tunneling)
 {
  u32 cd_cmd, cd_tso_len, cd_mss;
+ struct ipv6hdr *ipv6h;
  struct tcphdr *tcph;
  struct iphdr *iph;
  u32 l4len;
  int err;
- struct ipv6hdr *ipv6h;
 
  if (!skb_is_gso(skb))
   return 0;
 
- if (skb_header_cloned(skb)) {
-  err = pskb_expand_head(skb, 0, 0, GFP_ATOMIC);
-  if (err)
-   return err;
- }
+ err = skb_cow_head(skb, 0);
+ if (err < 0)
+  return err;
 
  if (protocol == htons(ETH_P_IP)) {
   iph = skb->encapsulation ? inner_ip_hdr(skb) : ip_hdr(skb);
-- 
1.7.1